localstack-core/localstack/utils/lambda_debug_mode/lambda_debug_mode_config.py

```diff
@@ -0,0 +1,198 @@
+        # Constructs yaml bindings, whilst checking for duplicate mapping key definitions, raising a
+        # MarkedYAMLError when one is found.
+        mapping = dict()
+        for key_node, value_node in node.value:
+            key = loader.construct_object(key_node, deep=deep)
+            if key in mapping:
+                # Create a MarkedYAMLError to indicate the duplicate key issue
+                raise MarkedYAMLError(
+                    context="while constructing a mapping",
+                    context_mark=node.start_mark,
+                    problem=f"found duplicate key: {key}",
+                    problem_mark=key_node.start_mark,
+                )
+            value = loader.construct_object(value_node, deep=deep)
+            mapping[key] = value
+        return mapping
+
+
+def from_yaml_string(yaml_string: str) -> Optional[LambdaDebugModeConfig]:
+    try:
+        data = yaml.load(yaml_string, _SafeLoaderWithDuplicateCheck)
+    except yaml.YAMLError as yaml_error:
+        LOG.error(
+            f"Could not parse yaml lambda debug mode configuration file due to: {str(yaml_error)}"
+        )
+        data = None
+    if not data:
+        return None
+    config = LambdaDebugModeConfig(**data)
+    return config
+
+
+def validate_lambda_debug_mode_config(config: LambdaDebugModeConfig) -> None:
+    _validate_lambda_debug_mode_config(
+        validation_state=_LambdaDebugModeConfigValidationState(), config=config
+    )
+
+
+def _validate_lambda_debug_mode_config(
+    validation_state: _LambdaDebugModeConfigValidationState, config: LambdaDebugModeConfig
+):
+    config_functions = config.functions
+    lambda_arns = list(config_functions.keys())
+    for lambda_arn in lambda_arns:
+        qualified_lambda_arn = _to_qualified_lambda_function_arn(lambda_arn)
+        if lambda_arn != qualified_lambda_arn:
+            if qualified_lambda_arn in config_functions:
+                raise DuplicateLambdaDebugConfig(
+                    lambda_arn_debug_config_first=lambda_arn,
+                    lambda_arn_debug_config_second=qualified_lambda_arn,
+                )
+            config_functions[qualified_lambda_arn] = config_functions.pop(lambda_arn)
+
+    for lambda_arn, lambda_debug_config in config_functions.items():
+        _validate_lambda_debug_config(
+            validation_state=validation_state, lambda_debug_config=lambda_debug_config
+        )
+
+
+def _to_qualified_lambda_function_arn(lambda_arn: Arn) -> Arn:
+    # Returns the $LATEST qualified version of a structurally unqualified version of a lambda Arn iff this
+    # if detected to be structurally unqualified. Otherwise, it returns the given string.
+    if not lambda_arn:
+        return lambda_arn
+    lambda_arn_parts = lambda_arn.split(":")
+    lambda_arn_parts_len = len(lambda_arn_parts)
+
+    # The arn is qualified and with a non-empy qualifier.
+    is_qualified = lambda_arn_parts_len == 8
```

Comment:
You could use the `api_utils.function_locators_from_arn` which will return a tuple of `(name, qualifier, account, region)` from a different schema of ARN (qualified, unqualified, full, partial, just function name, etc).

So your check could be:

```python
_, qualifier, *_ = api_utils.function_locators_from_arn(arn) 
is_arn_qualified = qualifier != None
# do stuff with the above ...
```

Reply:
Keep in mind until the above PR is merged in, an invalid/unmatched ARN will return a single `None` (not a tuple) which could break some things.

Reply:
I would choose to keep the logic of this function I have added. This logic should add the qualifier only to arns that contain all other parts. Although we don't parse or validate each part, for the majority of cases, we'd be adding or not a qualifier to correct arns. Unfortunately, `function_locators_from_arn` is not strict enough with identifying the parts. for example for `arn:aws:lambda:eu-central-1:000000000000:function` it selects `function` as the name, and null as the qualifier. Not only is it not correct (the function name is not actually given), but it also means that this logic would continue to produce `arn:aws:lambda:eu-central-1:000000000000:function:$LATEST`. Which seems a bit unfortunate. `_to_qualified_lambda_function_arn` would instead raise an UnknownLambdaArnFormat in this case, which would help the user getting the arn fixed (adding the name and/or the qualifier). `function_locators_from_arn` would is also too compliant with arns ending with empty qualifiers `arn:aws:lambda:eu-central-1:000000000000:function:functionname:`. Please do correct me if I have missing something! Otherwise, I think `_to_qualified_lambda_function_arn` is specific enough to this workflow that can be left as is. At the same time, I am not sure of the consequences of adding stricter checks to `function_locators_from_arn`.

Reply:
Just to clarify, it's actually AWS that interprets
```
arn:<partition>:lambda:<region>:≤account-id>:function
```
as 
```
arn:<partition>:lambda:<region>:<account-id>:function:function
```

So the `function_locators_from_arn` is keeping parity 🫠 

I think what you're saying is fair, and I given how much effort the additional validations required, I am also quite pro tackling this via string splitting over regex (where possible).

With that said, would you consider adding the above to `api_utils`? There are a couple places where I believe it could be useful in the provider.

Reply:
Thanks for the input! I'll reviews whether that "naming option" is something we should also support in this config file